core/app/[locale]/(default)/product/[slug]/_components/wishlist-button/form.tsx

```diff
@@ -0,0 +1,50 @@
+import { getTranslations } from 'next-intl/server';
+import { SearchParams } from 'nuqs';
+
+import { Streamable } from '@/vibes/soul/lib/streamable';
+
+import { wishlistAction } from '../../_actions/wishlist-action';
+import { SearchParamsRouterRefresh } from '../search-params-router-refresh';
+
+import { AddToNewWishlistModal } from './add-to-new-wishlist-modal';
+
+interface Props {
+  formId: string;
+  productId: number;
+  productSku: Streamable<string>;
+  searchParams: Promise<SearchParams>;
+}
+
+// Since the wishlist button lives inside of the ProductDetailsForm, we need a separate/detached form
+// to handle the actions through. The form ID is provided to the 'form' attribute of the wishlist menu items.
+// The Modal also contains a form, which needs to be detached.
+export const WishlistButtonForm = async ({
+  formId,
+  productId,
+  productSku,
+  searchParams,
+}: Props) => {
+  const t = await getTranslations('Account.Wishlists');
```

Comment:
This is a 🍹 but since these labels are not for the account page, I would re-define them under Product.